### DISCUSSION TITLE
Getting unable to verify first certificate after upgrading to 31.15.0

### DISCUSSION THREAD

**Discussion by ssharma2089** [renovatebot/renovate#13383] [Request Help] [answered]
I am upgrading renovate v28.10.1 to 31.15.0. Created renovate image by using custom root ca certificate. Followed the same process while creating new image. 
After using new image(31.15.0) in kubernetes deployment, getting a unable to verify a first certificate. Nothing has changed in the kubernetes configuration. However using a old image(28.10.1) is working.

DEBUG: Using RE2 as regex engine
DEBUG: Parsing configs
DEBUG: Checking for config file in /opt/renovate/config.js
DEBUG: Converting GITHUB_COM_TOKEN into a global host rule
DEBUG: File config
       "config": {
         "platform": "bitbucket-server",
         "username": "xxxxxx",
         "password": "xxxxxx",
         "onboarding": true,
         "token": "xxxx",
         "gitAuthor": "Renovate Bot <bot@renovateapp.com>",
         "repositories": ["MAD/xxxx"]
       }
DEBUG: CLI config
       "config": {}
DEBUG: Env config
       "config": {
         "hostRules": [
           {"hostType": "github", "matchHost": "github.com", "token": "xxxx"}
         ],
         "baseDir": "/tmp/renovate/",
         "platform": "bitbucket-server",
         "endpoint": "https://git.xxx.xxx/",
         "autodiscover": false
       }
DEBUG: Combined config
       "config": {
         "platform": "bitbucket-server",
         "username": "xxxxx",
         "password": "xxxx",
         "onboarding": true,
         "token": "xxxx",
         "gitAuthor": "Renovate Bot <bot@renovateapp.com>",
         "repositories": ["MAD/xxxx"],
         "hostRules": [
           {"hostType": "github", "matchHost": "github.com", "token": "xxxx"}
         ],
         "baseDir": "/tmp/renovate/",
         "endpoint": "https://git.xxx.xxx/",
         "autodiscover": false
       }
DEBUG: Adding trailing slash to endpoint
DEBUG: Found valid git version: 2.34.1
DEBUG: Using configured gitAuthor (Renovate Bot <bot@renovateapp.com>)
DEBUG: Adding password authentication for git.xxx.xxx to hostRules
DEBUG: Adding token authentication for git.xxx.xxx to hostRules
DEBUG: Using configured baseDir: /tmp/renovate/
DEBUG: Using cacheDir: /tmp/renovate/cache
DEBUG: Initializing Renovate internal cache into /tmp/renovate/cache/renovate/renovate-cache-v1
DEBUG: Commits limit = null
DEBUG: Clearing hostRules
DEBUG: Adding token authentication for github.com to hostRules
DEBUG: Adding password authentication for git.xxx.xxx to hostRules
DEBUG: Adding token authentication for git.xxx.xxx to hostRules
 INFO: Repository started (repository=MAD/xxxx)
       "renovateVersion": "31.15.0"
DEBUG: Using localDir: /tmp/renovate/repos/bitbucket-server/MAD/xxxx (repository=MAD/xxxx)
DEBUG: initRepo("{
         "repository": "MAD/xxxx"
       }") (repository=MAD/xxxx)
DEBUG: Unknown Bitbucket initRepo error (repository=MAD/xxxx)
       "err": {
         "name": "RequestError",
         "code": "UNABLE_TO_VERIFY_LEAF_SIGNATURE",
         "timings": {
           "start": 1641376080582,
           "socket": 1641376080584,
           "lookup": 1641376080586,
           "connect": 1641376080588,
           "error": 1641376080600,
           "phases": {"wait": 2, "dns": 2, "tcp": 2, "total": 18}
         },
         "message": "unable to verify the first certificate",
         "stack": "RequestError: unable to verify the first certificate\n    at ClientRequest.<anonymous> (/usr/src/app/node_modules/got/dist/source/core/index.js:962:111)\n    at Object.onceWrapper (events.js:520:26)\n    at ClientRequest.emit (events.js:412:35)\n    at ClientRequest.emit (domain.js:475:12)\n    at ClientRequest.origin.emit (/usr/src/app/node_modules/@szmarczak/http-timer/dist/source/index.js:43:20)\n    at TLSSocket.socketErrorListener (_http_client.js:475:9)\n    at TLSSocket.emit (events.js:400:28)\n    at TLSSocket.emit (domain.js:475:12)\n    at emitErrorNT (internal/streams/destroy.js:106:8)\n    at emitErrorCloseNT (internal/streams/destroy.js:74:3)\n    at processTicksAndRejections (internal/process/task_queues.js:82:21)\n    at TLSSocket.onConnectSecure (_tls_wrap.js:1515:34)\n    at TLSSocket.emit (events.js:400:28)\n    at TLSSocket.emit (domain.js:475:12)\n    at TLSSocket._finishInit (_tls_wrap.js:937:8)\n    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:709:12)",
         "options": {
           "headers": {
             "user-agent": "RenovateBot/31.15.0 (https://github.com/renovatebot/renovate)",
             "x-atlassian-token": "no-check",
             "authorization": "***********",
             "accept": "application/json",
             "accept-encoding": "gzip, deflate, br"
           },
           "url": "https://git.xxx.xxx/rest/api/1.0/projects/MAD/repos/xxxx",
           "hostType": "bitbucket-server",
           "username": "",
           "password": "",
           "method": "GET",
           "http2": false
         }
       }

**Comment by viceice** [comment, marked as answer]
1. Please format your logs
2. We can't realy help, as we don't know how you build your custom image
3. Checkout our offical docs for image customizing
    * https://github.com/containerbase/buildpack/blob/main/docs/custom-base-image.md
    * https://github.com/containerbase/buildpack/blob/main/docs/custom-root-ca.md

**Reply by ssharma2089** [reply]
Sorry. It's my fault. I followed the right steps to create the image, however I provided an invalid certificate. It's presently operational.

To pull the images, I'm using Harbor Registry. Some of the images are specific to harbor projects, while others images use proxy cache to pull images from quay.io, ghcr.io, etc. The following are some examples of images that are specified in deployment.

1. registry.xxx.xxx/quay.io/strimzi/kafka:0.25.0-kafka-2.8.0
2. selenium/maven-chrome:latest
What should I include in the renovate config to generate a new release/pull request for that?

**Reply by rarkins** [reply]
This discussion is already resolved

**Reply by viceice** [reply]
Please open a new discussion for your new question
